docs/contributing/05-gh-actions.md

```diff
@@ -2,205 +2,118 @@
-
-### gaurav-nelson/github-action-markdown-link-check
-
-- Repository: <https://github.com/gaurav-nelson/github-action-markdown-link-check>
-- Marketplace: <https://github.com/marketplace/actions/markdown-link-check>
-- `markdown-link-check` repository: <https://github.com/tcort/markdown-link-check>
-
-The action runs a shell script which installs and runs a Node.js module called markdown-link-check. The [version of the Node.js module is tagged](https://github.com/gaurav-nelson/github-action-markdown-link-check/blob/46e442156b8161bfd0913357c7a411e0e610d2ad/entrypoint.sh#L11).
-
-#### Rating
-
-- openSSF Scorecard rating (16.08.2023)
-  - [gaurav-nelson/github-action-markdown-link-check](./scorecard-ratings/gaurav-nelson_github-action-markdown-link-check.json): 4.4/10
-  - [tcort/markdown-link-check](./scorecard-ratings/tcort_markdown-link-check.json): 4.6/10
-
-#### Versions
-
-- gaurav-nelson/github-action-markdown-link-check@v1
-- gaurav-nelson/github-action-markdown-link-check@d53a906aa6b22b8979d33bc86170567e619495ec
-
-### slackapi/slack-github-action
-
-- Repository: <https://github.com/slackapi/slack-github-action>
-- Marketplace: <https://github.com/marketplace/actions/slack-send>
+## GitHub Actions Security Best Practices
 
-#### Ratings
+Even though the usage of GitHub Actions is restricted, some threats remain. The responsibility to target these threats is on the developer using GitHub Actions.
 
-- openSSF Scorecard rating (16.08.2023)
-  - [slackapi/slack-github-action](./scorecard-ratings/slackapi_slack-github-action.json): 6.1 / 10
+### Security review
 
-#### Versions
+The code of the action has to be reviewed to identify suspicious parts of the code. It should be especially checked if the Action is processing secrets beside the expected usage and what kind of modification the Action does on Github resources.
 
-- slackapi/slack-github-action@e28cf165c92ffef168d23c5c9000cffc8a25e117 # v1.24.0,
+### Set the minimal scope for credentials
 
-## Not allowed GitHub Actions
+- GITHUB_TOKEN has read and write permissions
+- Must be restricted to the least minimum
+- Also valid for all other credentials used by the Github Action
 
-### shmokmt/actions-setup-github-comment
+### Do NOT use `pull_request_target` event
 
-The GitHub Action is only a wrapper to run a go tool called [github-comment](https://github.com/suzuki-shunsuke/github-comment). It is downloaded during runtime by the Action.
+- Avoid using `pull_request_target` if the workflow doesn't need write repository permissions and doesn't use any repository secrets. They can simply use the `pull_request` trigger instead.
+- Assign repository privileges only where needed explicitly through pull_request and workflow_run
+  - Handle untrusted code via the `pull_request` trigger so that it is isolated in an unprivileged environment.This workflow started with the `pull_request` trigger should then store any results like code coverage or failed/passed tests in artifacts and exit.
```

Comment:
```suggestion
  - Handle untrusted code via the `pull_request` trigger so that it is isolated in an unprivileged environment. This workflow started with the `pull_request` trigger should then store any results like code coverage or failed/passed tests in artifacts and exit.
```